network-store-iidm-impl/src/test/java/com/powsybl/network/store/iidm/impl/VoltageLevelSetVTest.java

```diff
@@ -0,0 +1,336 @@
+            .setNode2(14)
+            .add();
+        vl1.getNodeBreakerView().newSwitch()
+            .setId("VL1SW10")
+            .setKind(SwitchKind.BREAKER)
+            .setRetained(true)
+            .setOpen(false)
+            .setNode1(13)
+            .setNode2(14)
+            .add();
+
+        return (VoltageLevelImpl) vl1;
+    }
+
+    private static final Map<String, String> BBVL_CONFIGUREDBUS_TO_BUSVIEWBUS = new TreeMap<>(Map.of(
+            "BUS1", "VL2_0",
+            "BUS2", "VL2_0",
+            "BUS3", "VL2_1",
+            "BUS4", "VL2_2",
+            "BUS5", ""
+    ));
+
+    private static final Map<String, List<String>> BBVL_BUSVIEWBUS_TO_CONFIGUREDBUS = invertMap(BBVL_CONFIGUREDBUS_TO_BUSVIEWBUS);
+
+    private static VoltageLevelImpl createBusBreaker(Network network) {
+        // busbreaker topology voltage level
+        VoltageLevel vl2 = network.newVoltageLevel()
+            .setId("VL2")
+            .setNominalV(225.0)
+            .setTopologyKind(TopologyKind.BUS_BREAKER)
+            .add();
+        vl2.getBusBreakerView().newBus()
+            .setId("BUS1")
+            .add();
+        vl2.getBusBreakerView().newSwitch()
+            .setId("VL2SW1")
+            .setOpen(false)
+            .setBus1("BUS1")
+            .setBus2("BUS2")
+            .add();
+        vl2.getBusBreakerView().newBus()
+            .setId("BUS2")
+            .add();
+        vl2.getBusBreakerView().newSwitch()
+            .setId("VL2SW2")
+            .setOpen(true)
+            .setBus1("BUS2")
+            .setBus2("BUS3")
+            .add();
+        vl2.getBusBreakerView().newBus()
+            .setId("BUS3")
+            .add();
+        vl2.getBusBreakerView().newBus()
+            .setId("BUS4")
+            .add();
+        vl2.getBusBreakerView().newBus()
+            .setId("BUS5")
+            .add();
+
+        // loads to avoid pruning
+        vl2.newLoad().setId("VL2L1").setBus("BUS1").setP0(1).setQ0(1).add().getTerminal().setP(0.001).setQ(1);
+        vl2.newLoad().setId("VL2L2").setBus("BUS3").setP0(1).setQ0(1).add().getTerminal().setP(0.001).setQ(1);
+        vl2.newLoad().setId("VL2L3").setBus("BUS4").setP0(1).setQ0(1).add().getTerminal().setP(0.001).setQ(1);
+        vl2.newLoad().setId("VL2L4").setConnectableBus("BUS5").setP0(1).setQ0(1).add().getTerminal().setP(0.001).setQ(1);
+        // BUS2 has no load it will be pruned
+
+        return (VoltageLevelImpl) vl2;
+    }
+
+    private void testSetVMultipleBusAcrossViews(Supplier<VoltageLevel> supplier, Map<String, String> busBreakerViewBusToBusViewBus, Map<String, List<String>> busViewBusToBusBreakerViewBus) {
+
+        VoltageLevel vl = supplier.get();
+        for (Map.Entry<String, String > entry : busBreakerViewBusToBusViewBus.entrySet()) {
+            String busbreakerviewbusid = entry.getKey();
+            String busviewbusid = entry.getValue();
+
+            for (Bus bbvb : vl.getBusBreakerView().getBuses()) {
```

Comment:
we need to refactor these test to allow testing with or without already calculated buses. Here because we force getBuses before the tests, we always run with calculated bus already computed